### ISSUE TITLE
unable to connect ESP-01 with UNO

### ISSUE THREAD

**Issue by Bwanna** [jeelabs/esp-link#454]
I've tried to follow everything exactly, but can't seem to make this work.  Trying to create a wireless FTDI adapter to use instead of a wired USB FTDI.
-Have successfully installed ESP-Link and have it connect to my network and able to interact with it using the onboard HTML server.
![image](https://user-images.githubusercontent.com/25287711/62009376-70c2c200-b12c-11e9-84a2-33973a7e8f6f.png)

-Connected ESP-01 to UNO using the following pinout:
ESP-01    UNO
Vcc           5v
GND         GND
TX             RX
RX             TX
GPIO0       RST

-Arduino IDE Setup with the following:
![image](https://user-images.githubusercontent.com/25287711/62009413-dca52a80-b12c-11e9-8d81-8eee8e28c444.png)

-Uploading results in the following error msg:
![image](https://user-images.githubusercontent.com/25287711/62009424-137b4080-b12d-11e9-899c-d5c4528132f9.png)

Can you provide any suggestions on what I've don't incorrectly?

**Comment by uzi18** [comment]
how did you connected it to uno? is it original uno or clone?

**Comment by Bwanna** [comment]
This is a clone UNO (sunflower)
-Connected ESP-01 to UNO using the following pinout:
ESP-01 -> UNO
Vcc -> 5v
GND ->GND
TX -> RX
RX -> TX
GPIO0 -> RST

I'd like to make a universal type of adapter that can plug into any FTDI of an MCU and work wirelessly.  Is that possible?

**Comment by uzi18** [comment]
did you tried to use console subpage on esp-link webserver?
what exactly is not working for you?

**Comment by Bwanna** [comment]
> did you tried to use console subpage on esp-link webserver?
> what exactly is not working for you?

You'll notice in the error msg in the screen capture above.  That's all I have and not sure where to go with it.  The UNO does have blink.ino currently loaded and it 'seems' that when the "Resetting the board" occurs that the UNO stops blinking.  I also notice on the ESP-01 that the blue light occasionally flickers for a little bit right after this... and then the UNO go back to blinking and the error msg is shown in the IDE.

Does the type of bootloader make a difference?  Just guessing, but since it's a UNO clone (Sunflower) does that make a difference?  

Re: 'subpage'.  I think you mean the uC interface?  Yes, I tried that and don't get any response from the serial monitor.

**Comment by uzi18** [comment]
ok I get it now, not sure if arduinoOta works with esp-link, something was changed in meantime in arduino ide
and error is "out of sync" , so looks like esp-link does not support something here

**Comment by Bwanna** [comment]
I thought arduinoOta was for uploading sketches **TO** the ESP8266, and not for uploading **THROUGH** this to the uC it is connected?
According to the ESP-Link info (https://github.com/jeelabs/esp-link) it's supposed to provide that transparent interface, in essence, act just like a wired USB FTDI connection would.

I was interpreting the 'out of sync' error as an issue with the ESP8266 not being able to setup the serial link with the UNO.  Is that what you mean?

**Comment by uzi18** [comment]
yes out of sync is usually related to avrdude and bootloader in uno, but your arduino ide launched arduinoOta so for me it is related  to arduinoOta-esplink communication, need to check how arduinoota works

**Comment by Bwanna** [comment]
Are you saying that the IDE launched arduinoOta because I selected the "Arduino Uno WiFi" ?  

The instructions on the ESP-Link stated using "Arduino Uno WiFi" to get the ESP-01 IP in the port selection.  Still learning here, but I would have thought since the ESP-Link is supposed to be just a 'pass-through' that the IDE board selection would correlate to the uC that the ESP-01 was connected as an FTDI.  In this case an UNO.
